controllers/subscription.ts

```diff
@@ -0,0 +1,72 @@
+import { CustomRequest, CustomResponse } from "../types/global";
+const { addOrUpdate } = require("../models/users");
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const nodemailer = require("nodemailer");
+const config = require("config");
+const emailSubscriptionCredentials = config.get("emailSubscriptionCredentials");
+
+export const subscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const { email, phoneNumber } = req.body;
+    const userId = req.userData.id;
+    await addOrUpdate(
+      {
+        phoneNumber,
+        email,
+        isSubscribed: true,
+      },
+      userId
+    );
+    return res.status(201).json({
```

Comment:
- are we creating a new resource here?

Reply:
> yes, we are adding users email and phone number here.

is this what we decided

Reply:
Yes sir, we decided to add email and phone number in user collection.

Reply:
@tejaskh3 does this not just update the phone number, email, and isSubscribed status in the user collection? if so how are we creating a new resource here?

Reply:
yes we are creating because they were not present their earlier